server/schedulers/utils.go

```diff
@@ -36,61 +36,102 @@ const (
 	minTolerantSizeRatio    float64 = 1.0
 )
 
-func shouldBalance(cluster opt.Cluster, source, target *core.StoreInfo, region *core.RegionInfo, kind core.ScheduleKind, opInfluence operator.OpInfluence, scheduleName string) (shouldBalance bool, sourceScore float64, targetScore float64) {
+type balancePlan struct {
+	kind        core.ScheduleKind
+	cluster     opt.Cluster
+	opInfluence operator.OpInfluence
+
+	source *core.StoreInfo
+	target *core.StoreInfo
+	region *core.RegionInfo
+
+	sourceScore float64
+	targetScore float64
+}
+
+func newBalancePlan(kind core.ScheduleKind, cluster opt.Cluster, opInfluence operator.OpInfluence) *balancePlan {
+	return &balancePlan{
+		kind:        kind,
+		cluster:     cluster,
+		opInfluence: opInfluence,
+	}
+}
+
+func (p *balancePlan) GetOpInfluence(storeID uint64) int64 {
+	return p.opInfluence.GetStoreInfluence(storeID).ResourceProperty(p.kind)
+}
+
+func (p *balancePlan) SourceStoreID() uint64 {
+	return p.source.GetID()
+}
+
+func (p *balancePlan) SourceStoreLabel() string {
```

Comment:
The function is a little bit misleading since we have a store label actually but with a different meaning.

Reply:
good point.